solr/core/src/java/org/apache/solr/filestore/DistribFileStore.java

```diff
@@ -180,23 +184,35 @@ private boolean fetchFileFromNodeAndPersist(String fromNode) {
 
       ByteBuffer metadata = null;
       Map<?, ?> m = null;
+
+      InputStream is = null;
+      var solrClient = coreContainer.getDefaultHttpSolrClient();
+
       try {
+        GenericSolrRequest request = new GenericSolrRequest(GET, "/node/files" + getMetaPath());
+        request.setResponseParser(new InputStreamResponseParser(null));
+        var response = solrClient.requestWithBaseUrl(baseUrl, request::process).getResponse();
+        is = (InputStream) response.get("stream");
         metadata =
-            Utils.executeGET(
-                coreContainer.getUpdateShardHandler().getDefaultHttpClient(),
-                baseUrl + "/node/files" + getMetaPath(),
-                Utils.newBytesConsumer((int) MAX_PKG_SIZE));
+            Utils.newBytesConsumer((int) MAX_PKG_SIZE).accept((InputStream) response.get("stream"));
         m = (Map<?, ?>) Utils.fromJSON(metadata.array(), metadata.arrayOffset(), metadata.limit());
```

Comment:
Please replace needless use of the special InputStreamResponseParser with a standard JsonMapResponseParser.  Perhaps don't even do that; we don't have to use JSON, I think; Solr will by default negotiate the format and get you a NamedList. (Map like thing).

Reply:
@iamsanjay you said at the meetup that this doesn't work sometimes.  I'm not surprised but I'd like to hear more about specifically why it doesn't work for particular callers that we might like to use, such as here.

Reply:
I now understand that we must do this because we need the metadata in ByteBuffer form to persist it later, therefore we need the InputStream.